roles/kube_lvm_storage/templates/local-pv.yml.j2

```diff
@@ -24,3 +12,11 @@ spec:
   storageClassName: '{{ item.value.storage_class }}'
   local:
     path: '{{ item.value.host_path }}/{{ item.value.volume.uuid }}'
+  nodeAffinity:
+    required:
+      nodeSelectorTerms:
+      - matchExpressions:
+        - key: kubernetes.io/hostname
+          operator: In
+          values:
+          - {{ ansible_nodename }}
```

Comment:
Whilst in this case it likely doesn't make a difference (`123` would not be a valid `ansible_nodename` anyway), consider quoting templated strings in YAML.